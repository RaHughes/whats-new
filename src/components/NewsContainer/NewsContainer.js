  
import React from 'react';
import './NewsContainer.scss';
import NewsArticle from '../NewsArticle/NewsArticle.js'

const NewsContainer = ({articles}) => {
    
    const newsArticles = articles.map(news => {
      return (<NewsArticle
        key={(Date.now() + Math.random())}
        img={news.img}
        headline={news.headline}
        description={news.description}
        url={news.url}
        />)
    });
  
    return (
       <section id="news-container">
         {newsArticles}
       </section>
     );
   }

export default NewsContainer