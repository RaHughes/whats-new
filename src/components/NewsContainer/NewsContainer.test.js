import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';

describe('NewsContainer', ()=> { 
  it('renders without crashing', () => {
    const testArticles = [{
      key: 1,
      img: 'this1',
      headling: 'healdine1',
      description: 'description1',
      url:'www.news1.com'
    }, 
    {
      key: 2,
      img: 'this2',
      headling: 'healdine2',
      description: 'description2',
      url: 'www.news2.com'
    }, 
    {
      key: 3,
      img: 'this3',
      headling: 'healdine3',
      description: 'description3',
      url: 'www.news3.com'
    }]
    const wrapper = shallow(<NewsContainer
     />);
    
    expect(wrapper).toMatchSnapshot();
  });
});