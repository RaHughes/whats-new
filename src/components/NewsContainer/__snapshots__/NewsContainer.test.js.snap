// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer renders without crashing 1`] = `ShallowWrapper {}`;
